cmd/env.go

```diff
@@ -48,46 +55,48 @@ func init() {
 }
 
 var envCmd = &cobra.Command{
 	Use:   "env",
 	Short: `Manage ksonnet environments`,
-	Long: `An environment acts as a sort of "named cluster", allowing for commands like
-` + " `ks apply dev` " + `, which applies the ksonnet application to the 'dev cluster'.
-Additionally, environments allow users to cache data about the cluster it points
-to, including data needed to run 'verify', and a version of ksonnet-lib that is
-generated based on the flags the API server was started with (e.g., RBAC enabled
-or not).
-
-An environment contains no user-specific data (such as the private key
-often contained in a kubeconfig file), and
-
-Environments are represented as a hierarchy in the 'environments' directory of a
-ksonnet application. For example, in the example below, there are two
-environments: 'default' and 'us-west/staging'. Each contains a cached version of
-` + " `ksonnet-lib` " + `, and a` + " `spec.json` " + `that contains the server and server cert that
-uniquely identifies the cluster.
-
-    environments/
-      default/           [Default generated environment]
-        .metadata/
-          k.libsonnet
-          k8s.libsonnet
-          swagger.json
-        spec.json
-		default.jsonnet
-        params.libsonnet		
-      us-west/
-        staging/         [Example of user-generated env]
-          .metadata/
-            k.libsonnet
-            k8s.libsonnet
-            swagger.json
-          spec.json      [This will contain the API server address of the environment and other environment metadata]
-		  staging.jsonnet
-          params.libsonnet`,
+	Long: `
+An environment is a deployment target for your ksonnet app and its constituent
+components. You can use ksonnet to deploy a given app to *multiple* environments,
+such as ` + "`dev`" + ` and ` + "`prod`" + `.
+
+Intuitively, an environment acts as a sort of "named cluster", similar to a
+Kubernetes context. (Running ` + "`ks env add --help`" + ` provides more detail
+about the fields that you need to create an environment).
+
+**All of this environment info is cached in local files**. Environments are
+represented as a hierarchy in the ` + "`environments/`" + ` directory of a ksonnet app, like
+'default' and 'us-west/staging' in the example below.
+
+` + "```" + `
+├── environments
+│   ├── base.libsonnet
+│   ├── default                      // Default generated environment ('ks init')
+│   │   ├── .metadata
+│   │   │   ├── k.libsonnet
+│   │   │   ├── k8s.libsonnet
+│   │   │   └── swagger.json
+│   │   ├── main.jsonnet
+│   │   ├── params.libsonnet
+│   │   └── spec.json
+│   └── us-west
+│       └── staging                  // Example of user-generated env ('ks env add')
+│           ├── .metadata
+│           │   ├── k.libsonnet      // Jsonnet library with Kubernetes-compatible types and definitions
+│           │   ├── k8s.libsonnet
+│           │   └── swagger.json
+│           ├── main.libsonnet
```

Comment:
I think it's worth documenting what this file is since people confuse it with `params.libsonnet`.

Perhaps: // Main environment file that is expanded on `apply`, `delete`, etc. Add environment-specific components or sidecars here.